tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,339 @@
+	chartVersionPreUpgrade := cisbenchmarkChartPreUpgrade.ChartDetails.Spec.Chart.Metadata.Version
+	require.Contains(c.T(), versionsList[1:], chartVersionPreUpgrade)
+
+	c.chartInstallOptions.Version, err = client.Catalog.GetLatestChartVersion(charts.CISBenchmarkName, catalog.RancherChartRepo)
+	require.NoError(c.T(), err)
+	scanName := namegen.AppendRandomString("scan")
+
+	c.T().Log("Running a scan")
+
+	// Run the scan
+	err = runScan(client, scanName, c.cisProfileName, c.project.ClusterID)
+	require.NoError(c.T(), err)
+
+	require.NoError(c.T(), err)
+
+	c.T().Log("Upgrading cis-benchmark chart to the latest version")
+	err = charts.UpgradeCISBenchmarkChart(client, c.chartInstallOptions)
+	require.NoError(c.T(), err)
+
+	c.T().Log("Waiting for cis-benchmark chart deployments to have expected number of available replicas after upgrade")
+	err = extencharts.WatchAndWaitDeployments(client, c.project.ClusterID, charts.CISBenchmarkNamespace, metav1.ListOptions{})
+	require.NoError(c.T(), err)
+
+	c.T().Log("Waiting cis-benchmark chart DaemonSets to have expected number of available nodes after upgrade")
+	err = extencharts.WatchAndWaitDaemonSets(client, c.project.ClusterID, charts.CISBenchmarkNamespace, metav1.ListOptions{})
+	require.NoError(c.T(), err)
+
+	cisbenchmarkChartPostUpgrade, err := extencharts.GetChartStatus(client, c.project.ClusterID, charts.CISBenchmarkNamespace, charts.CISBenchmarkName)
+	require.NoError(c.T(), err)
+
+	c.T().Log("Comparing installed and desired cis-benchmark versions")
+	chartVersionPostUpgrade := cisbenchmarkChartPostUpgrade.ChartDetails.Spec.Chart.Metadata.Version
+	require.Equal(c.T(), c.chartInstallOptions.Version, chartVersionPostUpgrade)
+
+	scanName = namegen.AppendRandomString("scan")
+
+	c.T().Log("Running a scan")
+
+	// Run the scan
+	err = runScan(client, scanName, c.cisProfileName, c.project.ClusterID)
+	require.NoError(c.T(), err)
+}
+
+func getClusterScanTemplate() (string, error) {
+	readYamlFile, err := os.ReadFile("./resources/cis-profiles/cluster-scan-template.yaml")
+	return string(readYamlFile), err
+}
+
+func findScanIdx(data []v1.SteveAPIObject, scanName string) int {
+	for idx, scanReport := range data {
+		if scanReport.Name == scanName {
+			return idx
+		}
+	}
+	return -1
+}
+
+func checkScanReport(steveClient *v1.Client, scanName string) error {
+	scanReportList, err := steveClient.SteveType(ClusterScanReportType).List(nil)
+	if err != nil {
+		return err
+	}
+
+	scanReportIdx := findScanIdx(scanReportList.Data, scanName)
+
+	for idx, scanReport := range scanReportList.Data {
+		if strings.Contains(scanReport.Name, scanName) {
+			scanReportIdx = idx
+			break
+		}
+	}
+
+	if scanReportIdx < 0 {
+		return errors.New("scan report not found")
+	}
+
+	cisReportSpec := &ClusterScanReportSpec{}
+	scanReportSpec := scanReportList.Data[scanReportIdx].Spec
+	err = v1.ConvertToK8sType(scanReportSpec, cisReportSpec)
```

Comment:
can all of these non-public, non-test funcs get moved to cisbenchmark.go, or possibly make a file in v2/actions if you think this will get reused in future tests that aren't in the charts tests.

Reply:
agreed. if we can maintain just the tests in here and move everything else to the helper that would be more readable.